var/spack/repos/builtin/packages/hip/0009-Improve-compilation-without-git-repo-and-remove-compiler-rt-linkage-for-host_disabletests.4.5.0.patch

```diff
@@ -0,0 +1,101 @@
+diff --git a/hipamd/CMakeLists.txt b/hipamd/CMakeLists.txt
+index b1ab39e..f1459ea 100755
+--- a/hipamd/CMakeLists.txt
++++ b/hipamd/CMakeLists.txt
+@@ -87,7 +87,20 @@ string(REPLACE "-" ";" VERSION_LIST ${HIP_VERSION_PATCH_GITHASH})
+ list(GET VERSION_LIST 0 HIP_VERSION_PATCH)
+ set(HIP_VERSION_GITDATE 0)
+ 
+-find_package(Git)
++# only look for git when we have a git repo
++if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
++  find_package(Git)
++endif()
++
++# FIXME: Two different version strings used.
++
++set(HIP_PACKAGING_VERSION_PATCH "0")
++set(HIP_VERSION_GITDATE "0")
++set(HIP_VERSION_PATCH "0")
++set(HIP_VERSION_GITHASH "0")
++set(HIP_VERSION_PATCH_GITHASH "0")
++set (HIP_LIB_VERSION_PATCH "0")
```

Comment:
This whole set strikes me as peculiar.  Why are the versions all getting set to zero rather than a valid version?  If it doesn't matter, then perhaps this is fine, but if these actually end up in the output cmake config file that could cause some significant problems.

Reply:
these changes existed from the beginning of 3.5.0 ..i ported the patch changes to 4.5.0 ..also, there was a new additions HIP_VERSION_PATH_GITHASH and HIP_LIB_VERSION_PATCH. I think these variables were not set properly in spack and due to that final .so name had issues when it got generated in spack.